lib/rules/lint-no-invalid-role.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+function ERROR_MESSAGE_INVALID_ROLE(element) {
+  return `Use of presentation role on <${element}> detected. Semantic elements should not be used for presentation.`;
+}
+
+const ALLOWED_ELEMENTS = ['div', 'img', 'span', 'svg'];
```

Comment:
I'd like a spec link to explain why these specific elements are "allowed". I didn't see that in any of the links in the rule documentation (though maybe I just missed it).

Reply:
Sure, I'll look for some explicit links- my plan was to start a rule that covers common failure [92](https://www.w3.org/WAI/WCAG21/Techniques/failures/F92) with ones I already know for sure, and then expand from there.

Reply:
Hmm, but the implementation is quite different. The current implementation prevents using `presentation` for _any_ element other than this explicit list. That's a **ton** of elements (including ones that are custom defined by users e.g. `<my-awesome-thing></my-awesome-thing>` which should probably be treated like a `div`), and it makes me worried 😟

Reply:
@rwjblue so I do think that I should reverse this and do the other way that we talked about this morning, even though the list is longer, it _is_ a complete list of semantic elements. That way custom components wouldn't get caught up in the mix. What do you think?

Reply:
Yes, I think that would be much less risky.

Reply:
Going to close this issue and write it the other way.